Allow admins to set a limit on the length of usernames

happy-river opened the pull request:
Some of our users are using entire sentences for their usernames.  Our admins would like to be able to stop this behavior.  Add `site.username_max_length` to the admin site configuration panel, and enforce the limit at registration.

Polsaker commented:
Could you also set the `maxlength` attribute in the registration form? That'd make it a bit more user friendly

happy-river commented:
Updated it to use a `wtforms` `Validator` to check the username length, and added a description of the expected length toth e field title for users to see when they hover over the field.
